feat: add CalendarPdfExportButtons component with tests

Harishmakka opened the pull request:
- Adds dropdown-based Export to PDF button
- Handles export options for Daily, Weekly, Monthly, and Course
- Includes unit tests verifying UI behavior and handler calls

Harishmakka committed:
feat: add CalendarPdfExportButtons component with tests

- Adds dropdown-based Export to PDF button
- Handles export options for Daily, Weekly, Monthly, and Course
- Includes unit tests verifying UI behavior and handler calls

achow-conestogac commented:
these buttons don't do anything.

The whole point of the button is to export something to pdf.   if it doesn't do that, then it's not ready for review.

also, I want to see this component used.

Harishmakka committed:
feat(CalendarView): integrate CalendarPdfExportButtons for export functionality into CalendarView

achow-conestogac commented:
> Hi Professor,rightnow the component is used in CalendarView.tsx

i don't see it being used.

achow-conestogac commented on src/Evaluation/EvaluationCalendar/CalendarView/CalendarView.tsx:
```diff
@@ -1,12 +1,47 @@
-const CalendarView: React.FC<CalendarViewProps> = ({ evaluations }) => {
+const CalendarView: React.FC<CalendarViewProps> = ({ evaluations, viewMode }) => {
+  // Export handlers
+  const handleExportDaily = () => {
+    console.log("Exporting Daily View to PDF...");
+  };
+
+  const handleExportWeekly = () => {
+    console.log("Exporting Weekly View to PDF...");
+  };
+
+  const handleExportMonthly = () => {
+    console.log("Exporting Monthly View to PDF...");
+  };
+
+  const handleExportCourse = () => {
+    console.log("Exporting Entire Course to PDF...");
+  };
```
nonononononono

very bad.  mocking again.  i guess you have not developed the exporting functionality.

then this PR is far from ready to be merged.

Harishmakka committed:
Implement CalendarView PDF export integration and unit tests using real data

achow-conestogac commented:
there are existing concerns i have created which is not addressed.

there are also merge conflicts.

Roheeth05 committed:
This PR integrates PDF export functionality with CalendarView.

- PdfExportTypes and Tests
- PdfTemplates and Tests
- reportBuilder and Tests
- PdfGenerator and Tests
- Adds report generation from Evaluation[] → EvaluationReport
- Uses html2canvas + jsPDF to generate and download reports
- Supports Weekly and Monthly PDF exports

Harishmakka committed:
feat: integrate Calendar PDF export buttons and full export logic into CalendarView

- Added CalendarPdfExportButtons component to support export actions
- Implemented export handlers: handleExportDaily, Weekly, Monthly, Course
- Integrated PDF generation using buildEvaluationReport, EvaluationReportTemplate, and exportEvaluationReportToPDF
- Added hidden render container (#calendar-pdf) to support off-screen rendering for PDF export
- Export handlers filter evaluations based on current view and trigger download
- Updated the tests for PdfButtons
- Updated the tests for CalendarView
